common/network/uri_parser.c

```diff
@@ -16,6 +16,11 @@ bool uri_parse(char const *const uri, char *const scheme,
   size_t length = 0;
   size_t offset = 0;
   char *delim = NULL;
+  ip_version_t uri_case;
+  const char *possible_scheme[] = {PROTOCOL_TCP, PROTOCOL_UDP, PROTOCOL_HTTP,
```

Comment:
IMO the parser should only check for syntactic errors and fill fields but not validate the values of the tokens, please remove this part.